BUILDSCRIPT_MULTIPROC.bash44

```diff
@@ -1,20 +1,13 @@
 #!/bin/bash
 
-CONFIG="default"
-while [ $# -gt 0 ] ; do
-    if [ "$1" = "--config" ] && [ $# -gt 1 ] ; then
-        CONFIG="$2"
-        shift
-    fi
-    shift
-done
-
 CXX="${CXX:-g++}"
-CXXFLAGS="${CXXFLAGS:--Wall -Wextra -Wshadow -Wold-style-cast -Woverloaded-virtual -pedantic -O3 -flto}"
+CXXFLAGS="${CXXFLAGS:--Wall -Wextra -Wshadow -Wold-style-cast -Woverloaded-virtual -pedantic -O3}"
+CXXLTOFLAGS="${CXXLTOFLAGS:--flto}"
 CXXARCH="${CXXARCH:--march=native}"
 CXXSTD="${CXXSTD:--std=c++2a}"
 CXXFLAGSEXTRA="${CXXFLAGSEXTRA:-}"
-LDFLAGS="${LDFLAGS:--Wall -Wextra -Wl,-z,now -Wl,--as-needed -O3 -s -flto=auto}"
+if [[ $OSTYPE != "darwin"* ]]; then LDPLATFORMFLAGS="-Wl,-z,now -Wl,--as-needed -s"; fi
+LDFLAGS="${LDFLAGS:--Wall -Wextra -O3 -flto $LDPLATFORMFLAGS}"
```

Comment:
The `-flto=auto` flag isn't supported in many versions of GCC/Clang that are still frequently in use. For example it's not present in Clang 11, which is the default C/C++ compiler used for Darwin in `nixpkgs`. As such, I'd suggest just using `-flto`, since it's supported much more broadly, and as far as I can tell there isn't a significant downside. (Including `-flto` here is fine here even if it's on included in `$CXXFLAGS` above.)

The linker on Darwin doesn't support the `-z` and `--as-needed` flag, and it considers the `-s` flag obsolete. It would be nice if the build script excluded those flags automatically on Darwin systems, so that folks building on Darwin don't need to override `$LDFLAGS`.